src/inmanta/compiler/__init__.py

```diff
@@ -196,8 +196,15 @@ def compile(self) -> Tuple[List["Statement"], List["BasicBlock"]]:
 
         project.log_installed_modules()
 
+        # This lookup variable provides efficiency in the loop bellow:
+        # only call PluginMeta.clear once per module that is not loaded
```

Comment:
This sentence is not really correct. I think you are mixing up the loading of the Python module with the loading of the Inmanta module. 
```suggestion
        # This lookup variable provides efficiency in the loop bellow:
        # Make sure that `PluginMeta.clear` is called only once for each loaded Python module that is part of a non-loaded (non-imported) Inmanta module.
```